source/daplink/interface/main.c

```diff
@@ -213,6 +213,9 @@ __task void main_task(void)
     uint8_t thread_started = 0;
     // button state
     main_reset_state_t main_reset_button_state = MAIN_RESET_RELEASED;
+    // Initialize DAP locks.
+    util_assert(dap_lock_mutex_init());
```

Comment:
Functions with side effects should be kept out of asserts, so when asserts are turned off the program behavior does not change.